runs.py

```diff
@@ -120,6 +121,54 @@ def removals(self) -> list[str]:
         all_removals = self.neow_bonus.cards_removed + event_removals + store_removals
         return all_removals
 
+    @property
+    def character_streak(self) -> StreakInfo:
+        """Gets the run position in the character streak."""
+        return self._get_streak(True)
+
+    @property
+    def rotating_streak(self) -> StreakInfo:
+        """Gets the run position in the rotating streak."""
+        return self._get_streak(False)
+
+    def _get_streak(self, is_character_streak: bool) -> StreakInfo:
+        if self.won:
+            streak_total = 1
+            position_in_streak = 1
+
+            def loop_cached_runs(matched_key: str, should_iterate_position: bool):
+                nonlocal streak_total, position_in_streak
```

Comment:
dunno how we feel about this pattern. Wanted to be able to increment `streak_total` and `position_in_streak` without having to pass by reference and also not duplicate the loops of going forward and backward below.